src/Donations/LeftPanel/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { ReactElement, useContext, useState, useEffect } from "react";
+import LeftPanelContainer from "./LeftPanelContainer";
+import BackButton from "../../Common/BackButton";
+import { QueryParamContext } from "../../Layout/QueryParamContext";
+import LeftPanelHeader from "./LeftPanelHeader";
+import TopProjectBadge from "./TopProjectBadge";
+import LeftPanelInfo from "./LeftPanelInfo";
+
+function LeftPanel(): ReactElement {
+  const {
+    projectDetails,
+    donationID,
+    donationStep,
+    quantity,
+    paymentSetup,
+    currency,
+    contactDetails,
+    giftDetails,
+    isGift,
+    tenant,
+    frequency,
+    onBehalf,
+    onBehalfDonor,
+    isPlanetCashActive,
+    country,
+    callbackUrl,
+  } = useContext(QueryParamContext);
+
+  const [isMobile, setIsMobile] = useState(false);
+
+  const canShowBackButton =
+    isMobile && (callbackUrl.length > 0 || donationStep !== 0);
+  const canShowTopProjectBadge =
+    projectDetails !== null &&
+    projectDetails.purpose !== "planet-cash-signup" &&
+    projectDetails.isApproved &&
+    projectDetails.isTopProject;
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      if (window.innerWidth > 767) {
+        setIsMobile(false);
+      } else {
+        setIsMobile(true);
+      }
+    }
+  });
+
+  return (
+    <LeftPanelContainer
+      projectDetails={projectDetails}
+      donationStep={donationStep}
+      tenant={tenant}
+    >
+      <LeftPanelHeader>
+        {canShowBackButton && <BackButton backUrl={callbackUrl || "/"} />}
+        {canShowTopProjectBadge && <TopProjectBadge />}
+      </LeftPanelHeader>
+      {/* TODO - evaluate whether to send this info to LeftPanelInfo, or use context instead */}
```

Comment:
Do you mean for the `LeftPanelInfo` component? 

1. I want to avoid accessing/updating context in too many places, as that has the potential to create confusing code.  
2. The `LeftPanelInfo` component will be broken up into separate components and simplified in the future.  Have not decided the structure yet. At that point, we could decide to decide to use context within `LeftPanelInfo` instead of sending props ahead (for example, if `LeftPanelInfo` is a parent component instead of a container component). So this could change in the future.

Reply:
How could it be confusing, since context is the single source of truth for all its child components?